cli/src/main/java/com/devonfw/tools/ide/commandlet/ShellCommandlet.java

```diff
@@ -129,9 +131,37 @@ private int runCommand(String args) {
     String[] arguments = args.split(" ", 0);
     CliArguments cliArgs = new CliArguments(arguments);
     cliArgs.next();
+
+    if ("cd".equals(arguments[0])) {
+      return changeDirectory(cliArgs);
+    }
+
     return ((AbstractIdeContext) this.context).run(cliArgs);
   }
 
+  private int changeDirectory(CliArguments cliArgs) {
+    if (!cliArgs.hasNext()) {
+      Path homeDir = Paths.get(System.getProperty("user.home"));
```

Comment:
Not testable this way.
```suggestion
      Path homeDir = this.context.getUserHome();
```